app/controllers/locations_controller.rb

```diff
@@ -15,7 +15,147 @@ def index
+        else
+          redirect_to scheme_location_name_path(@scheme, @location)
+        end
+      else
+        render :postcode, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def local_authority
+    if params[:location].present?
+      @location.location_admin_district = params[:location][:location_admin_district]
+      @location.location_code = Location.local_authorities.key(params[:location][:location_admin_district])
+      if @location.valid?(:local_authority)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location)
+        else
+          redirect_to scheme_location_name_path(@scheme, @location)
+        end
+      else
+        render :local_authority, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def name
+    if params[:location].present?
+      @location.name = params[:location][:name]
+      if @location.valid?(:name)
+        @location.save!
+        case params[:referrer]
+        when "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location)
+        when "details"
+          redirect_to scheme_location_path(@scheme, @location)
+        else
+          redirect_to scheme_location_units_path(@scheme, @location)
+        end
+      else
+        render :name, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def units
+    if params[:location].present?
+      @location.units = params[:location][:units]
+      if @location.valid?(:units)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location)
+        else
+          redirect_to scheme_location_type_of_unit_path(@scheme, @location)
+        end
+      else
+        render :units, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def type_of_unit
+    if params[:location].present?
+      @location.type_of_unit = params[:location][:type_of_unit]
+      if @location.valid?(:type_of_unit)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location)
+        else
+          redirect_to scheme_location_mobility_standards_path(@scheme, @location)
+        end
+      else
+        render :type_of_unit, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def mobility_standards
+    if params[:location].present?
+      @location.mobility_type = params[:location][:mobility_type]
+      if @location.valid?(:mobility_type)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location)
+        else
+          redirect_to scheme_location_availability_path(@scheme, @location)
+        end
+      else
+        render :mobility_standards, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def availability
```

Comment:
this is now a manadatory field acording to the designs - it wasn't previously, is there anything we'll have to do for this?